### ISSUE TITLE
High CPU usage while mq is opened

### ISSUE THREAD

**Issue by jkuba** [mscdex/pmq#3]
I've modified the open/read example slightly to create a queue (if not exists), open it, and wait for an incoming message

``` JavaScript
var PosixMQ = require('pmq');
var readbuf, mq;

mq = new PosixMQ();
mq.on('messages', function() {
    var n;
    while((n = this.shift(readbuf)) !== false) {
        console.log('Received message (' + n + ' bytes): ' + readbuf.toString('utf8', 0, n));
        console.log('Messages left: ' + this.curmsgs);
    }
    this.unlink();
    this.close();
});

mq.open({ name: '/pmqtest', create: true, mode: '0777' });
readbuf = new Buffer(mq.msgsize);
```

While the application is waiting for a message, Node is showing 100% utilization of the CPU.

![image](https://f.cloud.github.com/assets/5039183/819057/571e503a-efa9-11e2-8ef1-d5bce9cebd40.png)

I am running Ubuntu 12.04 LTS and Node v0.10.6.

**Comment by delr** [comment]
same here. I was just going to post the same thing.. running node V 0.10.12, OpenSuse 12.3.
looks it starts polling after the message queue is opened.
is it possible to make it based on e.g. mq_notify ?

**Comment by mscdex** [comment]
This should be fixed in 07460d1. Please give it a try.

**Comment by jkuba** [comment]
Fix confirmed, Thanks!
